### PR TITLE
[INFINITY-2909] Verify self-resolution by default

### PR DESCRIPTION
**Pull request by benclarkwood**
Fix golint
Refactor resolution into components
Leverage refactor to verify task DNS => task IP
Add new flag "--self-resolve" to toggle new behavior of verified self resolution

### CODE REVIEW

File: sdk/bootstrap/main.go [hunk 1]
```diff
@@ -80,29 +85,40 @@ func parseArgs() args {
...
 	}
 
 	return args
 }
 
+func parseTaskHost() (string, error) {
+	taskName, taskNameOk := os.LookupEnv("TASK_NAME")
+	frameworkHost, frameworkHostOk := os.LookupEnv("FRAMEWORK_HOST")
+	if !taskNameOk || !frameworkHostOk {
+		return "", fmt.Errorf("Cannot determine TASK_NAME or FRAMEWORK_HOST")
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: Could have this return the message currently at L99/L219, and then have L99/L219 log whatever's in `err`

File: sdk/bootstrap/main.go [hunk 2]
```diff
@@ -356,20 +436,20 @@ func getContainerIPAddress() (string, error) {
...
 	if err != nil {
-		log.Fatalf("Failed to SET new LIBPROCESS_IP: ", err)
+		log.Fatalf("Failed to SET new LIBPROCESS_IP: %s", err)
 	}
 
-	if args.getTaskIp {
-		log.Printf("Printing new task IP: %s", pod_ip)
-		fmt.Printf("%s", pod_ip)
+	if args.getTaskIP {
+		log.Printf("Printing new task IP: %s", podIP)
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: This seems redundant, remove?

File: sdk/bootstrap/main.go [hunk 3]
```diff
@@ -393,6 +477,6 @@ func main() {
 		installDCOSCertIntoJRE()
 	}
 
-	log.Printf("Local IP --> %s", pod_ip)
+	log.Printf("Local IP --> %s", podIP)
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: Remove IMO, looks ugly in practice

File: sdk/bootstrap/main.go [hunk 4]
```diff
@@ -132,47 +197,62 @@ func waitForResolve(resolveHosts []string, resolveTimeout time.Duration) {
...
+
+func verifySelfResolution(podIP string, resolveTimeout time.Duration) {
+	taskHost, err := parseTaskHost()
+	if err != nil {
+		printEnv()
+		log.Printf("Missing required envvars to build task DNS address. " +
+			"Ensure that TASK_NAME and FRAMEWORK_HOST are both set or " +
+			"disable self resolution with --self-resolve=false")
+	}
+	log.Printf("Verifying that %s resolves to %s", taskHost, podIP)
```
Comment (nickbp) [inline sdk/bootstrap/main.go]: `Verifying that %s resolves` => `Waiting for %s to resolve`?

**Comment by nickbp** [comment]
Fails kafka integration due to `sdk_hosts.resolve_hosts()` attempting to run bootstrap via `task exec`:

```
2018/01/04 20:44:31 Missing required envvars to build task DNS address. Ensure that TASK_NAME and FRAMEWORK_HOST are both set or disable self resolution with --self-resolve=false
2018/01/04 20:44:31 Verifying that  resolves to 10.0.3.60
2018/01/04 20:44:31 Waiting for '' to resolve...
2018/01/04 20:49:31 Time ran out while resolving ''. Customize timeout with -resolve-timeout, or use -verbose to see attempts.
```

File: sdk/bootstrap/main.go [hunk 5]
```diff
@@ -132,47 +197,62 @@ func waitForResolve(resolveHosts []string, resolveTimeout time.Duration) {
...
+	if !timer.Stop() {
+		<-timer.C
+	}
+}
+
+func verifySelfResolution(podIP string, resolveTimeout time.Duration) {
+	taskHost, err := parseTaskHost()
+	if err != nil {
+		printEnv()
+		log.Printf("Missing required envvars to build task DNS address. " +
```
Comment (elezar) [inline sdk/bootstrap/main.go]: Is the intention to continue execution here? Should we exit or break?

File: sdk/bootstrap/main.go [hunk 6]
```diff
@@ -132,47 +197,62 @@ func waitForResolve(resolveHosts []string, resolveTimeout time.Duration) {
...
-				}
+	var timer *time.Timer
+	if resolveTimeout == 0 {
+		timer = nil
+	} else {
+		timer = time.NewTimer(resolveTimeout)
+	}
+
+	for {
+		resolvedIPs := resolveHost(taskHost, timer)
```
Comment (elezar) [inline sdk/bootstrap/main.go]: Why is this treated as such as special case? Is this not the same as calling `waitForResolve()` with a list containing the `taskHost` (with the modification that we need add return values from `waitForResolve()`?

File: sdk/bootstrap/main.go (hunk 1 above)
Comment (benclarkwood) [reply sdk/bootstrap/main.go]: The message is slightly different:

@99: "set resolve hosts"

@219: "Time ran out waiting to resolve X to Y"

File: sdk/bootstrap/main.go (hunk 5 above)
Comment (benclarkwood) [reply sdk/bootstrap/main.go]: Whoops. That should be a Fatalf.

File: sdk/bootstrap/main.go (hunk 6 above)
Comment (benclarkwood) [reply sdk/bootstrap/main.go]: I want the custom log lines around resolving to something specific.

**Comment by benclarkwood** [comment]
@nickbp @elezar A re-review when you have time please.

**Pull request merged**